test/e2e/specs/editor/plugins/block-directory.spec.js

```diff
@@ -0,0 +1,258 @@
+	author_block_count: 1,
+	author: 'No Author',
+	icon: 'block-default',
+	assets: [
+		'https://fake_url.com/block.js', // We will mock this.
+	],
+	humanized_updated: '5 months ago',
+	links: {},
+};
+
+const MOCK_INSTALLED_BLOCK_PLUGIN_DETAILS = {
+	plugin: 'block-directory-test-block',
+	status: 'active',
+	name: 'Block Directory',
+	plugin_uri: '',
+	author: 'No Author',
+	author_uri: '',
+	description: {
+		raw: 'This plugin is useful for the block.',
+		rendered: 'This plugin is useful for the block.',
+	},
+	version: '1.0',
+	network_only: false,
+	requires_wp: '',
+	requires_php: '',
+	text_domain: 'block-directory-test-block',
+	_links: {
+		self: [
+			{
+				href: '',
+			},
+		],
+	},
+};
+
+const MOCK_BLOCK2 = {
+	...MOCK_BLOCK1,
+	name: 'block-directory-test-block/secondary-block',
+	title: 'Block Directory Test Block - Pt Deux',
+	id: 'block-directory-test-secondary-block',
+};
+
+// Block that will be registered.
+const block = `( function() {
+	var registerBlockType = wp.blocks.registerBlockType;
+	var el = wp.element.createElement;
+
+	registerBlockType( '${ MOCK_BLOCK1.name }', {
+		title: 'Block Directory Test Block',
+		icon: 'hammer',
+		category: 'text',
+		attributes: {},
+		edit: function( props ) {
+			return el( 'p', null, 'Test Copy' );
+		},
+		save: function() {
+			return null;
+		},
+	} );
+} )();`;
+
+function matchUrl( requestUrl, urls ) {
+	return urls.some( ( url ) => requestUrl.href.includes( url ) );
+}
+
+test.describe( 'Block Directory', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should show an empty state when no plugin is found', async ( {
+		page,
+	} ) => {
+		// Return an empty list of plugins.
+		await page.route(
+			( url ) => matchUrl( url, SEARCH_URLS ),
+			async ( route, request ) => {
+				if ( request.method() === 'GET' ) {
+					await route.fulfill( {
+						json: [],
+					} );
+				} else {
+					await route.continue();
+				}
+			}
+		);
+
+		await page
+			.getByRole( 'toolbar', { name: 'Document tools' } )
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		const blockLibrary = page.getByRole( 'region', {
+			name: 'Block Library',
+		} );
+
+		/**
+		 * Search for the block via the inserter.
+		 * Be super weird so there won't be a matching block installed.
+		 */
+		await blockLibrary
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( '@$@@Dsdsdfw2$@' );
+
+		/**
+		 * The inserter can flash "No results found" before requesting results from the directory,
+		 * fulfilling the first assertion.
+		 * Waiting for these two elements ensures that the request was dispatched.
+		 */
+		await expect(
+			blockLibrary.locator( '.block-editor-inserter__no-results' )
+		).toHaveText( 'No results found.' );
+		await expect(
+			blockLibrary.locator( '.block-editor-inserter__tips' )
+		).toContainText( 'Interested in creating your own block?' );
```

Comment:
If those specific classes aren't critical, we could skip them and expect the text to be inside the `blockLibrary`:
```suggestion
		await expect( blockLibrary ).toContainText( 'No results found.' );
		await expect( blockLibrary ).toContainText( 'Interested in creating your own block?' );
```